soql-standalone-parser/src/main/scala/com/socrata/soql/parsing/RecursiveDescentParser.scala

```diff
@@ -1159,6 +1159,38 @@ abstract class RecursiveDescentParser(parameters: AbstractParser.Parameters = Ab
     }
   }
 
+  // param("var") or param(@table, "var")
+  private def param(reader: Reader, pos: Position): ParseResult[Expression] = {
+    reader.first match {
+      case TableIdentifier(ti) =>
+        reader.rest.first match {
+          case COMMA() =>
+            reader.rest.rest.first match {
+              case tokens.StringLiteral(varName) =>
+                reader.rest.rest.rest.first match {
+                  case RPAREN() =>
+                    ParseResult(reader.rest.rest.rest.rest, Hole.SavedQuery(HoleName(varName), Some(ti.drop(1)))(pos))
+                  case _ =>
+                    fail(reader.rest.rest.rest, RPAREN())
+                }
+              case _ =>
+                fail(reader.rest.rest, AStringLiteral)
+            }
+          case _ =>
+            fail(reader.rest, COMMA())
+        }
+      case tokens.StringLiteral(varName) =>
+        reader.rest.first match {
+          case RPAREN() =>
+            ParseResult(reader.rest.rest, Hole.SavedQuery(HoleName(varName), None)(pos))
```

Comment:
I thought we might be adding the inferred table name here, does that happen somewhere else?